python_bindings/PythonBindings.cc

```diff
@@ -29,11 +30,29 @@ template <typename T> void bindCsf(py::module &module, const char *name, const u
         }, py::arg("filename"));
 }
 
+template <typename T> void bindPermutation(py::module &m, const char *name){
+  m.def(name, [](py::array_t<T, py::array::c_style | py::array::forcecast> &array) {
+    // Check array dimensions and load the buffer into a pointer.
+    if (array.ndim() != 2) {
+      throw std::runtime_error("Input should be a 2D numpy array.");
+    }
+    py::buffer_info info = array.request();
+    int num_rows = info.shape[0];
+    int num_cols = info.shape[1];
+    T* M = static_cast<T*>(info.ptr);  // must be row-major format.
+    entropyPermutation<T>(M, num_rows, num_cols);
```

Comment:
this is nice! Doing this in place for no copies and handling all the numpy stuff